src/Form/TimesheetEditForm.php

```diff
@@ -202,12 +203,12 @@ function (FormEvent $event) {
                 /** @var Timesheet $data */
                 $data = $event->getData();
                 $duration = $data->getDuration();
-                $end = null;
-                if (null !== $duration) {
+                if (!empty($duration)) {
+                    $end = null;
```

Comment:
why? the next line sets $end, was that move by accident?

Reply:
I moved the setEnd() call to inside the if(), so it made sense to move the whole $end variable handling there too – this is to prevent us from overriding the end value if we don't alter it here. But you're absolutely right, this line is meaningless here and should be removed.